pkg/plugin/resources.go

```diff
@@ -171,26 +194,42 @@ func (app *App) handleReport(w http.ResponseWriter, req *http.Request) {
 	// So we need to rely on either service accounts or user provided API tokens to
 	// make requests to Grafana
 	case req.Header.Get(backend.CookiesHeaderName) != "":
+		ctxLogger.Debug("using cookies")
+
 		credential = client.Credential{
 			HeaderName:  backend.CookiesHeaderName,
 			HeaderValue: req.Header.Get(backend.CookiesHeaderName),
 		}
-	case req.Header.Get(backend.OAuthIdentityTokenHeaderName) != "":
+	case conf.Token != "":
+		ctxLogger.Debug("using configured token")
+
+		token := conf.Token
+
+		// support all token types, including non-bearer tokens
+		if !strings.Contains(token, " ") {
+			token = "Bearer " + token
+		}
```

Comment:
It is not clear. You mean to comment "Support only bearer tokens". Because in the code you are hard coding the `Bearer`.

Reply:
For example. if end-user enter `Basic YWRtaW46YWRtaW4=` here and it would not longer prefix with "Bearer". Thats what I mean with support non-bearer tokens.

Reply:
Ah, I understand the comment now. Cheers for the explanation. 

I guess we **should not** support Basic auth. Imagine a situation where a regular user makes a API request with basic auth (username and password). Since, this basic auth header will not be forwarded to plugin by Grafana, the plugin will use service account token to make API requests to generate report. In this case, the user might be able to generate reports on the dashboards that they do not have permissions. Does it make sense?

We should only allow users with Admin role to make API requests to generate report. Normally it is only admin users that can generate service account API tokens and so it is only admins who can interact with plugin in API mode. And regular users should **always** generate reports using browser sessions where RBAC is imposed either by the cookie or by token forwarded by `idForwarding`.

Do you agree?

Reply:
It is **only** admin users who can configure this token. And I am not sure if we should encourage them using Basic auth by passing token as `Basic YWRtaW46YWRtaW4=`.

Reply:
I have more the opinion to give admins a full flexibility. I understand your concerns. However an administrator has actively configure this to create an insecure environment. 

But I'm going to revert this, since its off-topic in context of this PR.